src/store/Conversations.js

```diff
@@ -43,6 +43,21 @@ export const conversations = createSlice({
       });
       conversationsAdapter.setAll(state, conversations);
     },
+    upsertChats: (state, action) => {
+      const conversations = action.payload;
+      const conversationsToUpdate = [];
+      conversations.forEach((conv) => {
+        if (state.entities[conv._id]) {
+          return;
```

Comment:
so you filter convs that are already exist in store. Hence it does not look like an upsert. It's rather insert